integration/embed_test.go

```diff
@@ -31,9 +35,113 @@ func magnitude[V float32 | float64](v []V) V {
 }
 
 func cosineSimilarity[V float32 | float64](v1, v2 []V) V {
+	mag1 := magnitude(v1)
+	mag2 := magnitude(v2)
+
+	if mag1 == 0 || mag2 == 0 {
+		return 0
+	}
+
 	return dotProduct(v1, v2) / (magnitude(v1) * magnitude(v2))
 }
 
+func euclideanDistance[V float32 | float64](v1, v2 []V) V {
+	if len(v1) != len(v2) {
+		return V(math.Inf(1))
+	}
+
+	var sum V = 0
+	for i := 0; i < len(v1); i++ {
+		diff := v1[i] - v2[i]
+		sum += diff * diff
+	}
+
+	return V(math.Sqrt(float64(sum)))
+}
+
+func manhattanDistance[V float32 | float64](v1, v2 []V) V {
+	if len(v1) != len(v2) {
+		return V(math.Inf(1))
+	}
+
+	var sum V = 0
+	for i := 0; i < len(v1); i++ {
+		sum += V(math.Abs(float64(v1[i] - v2[i])))
+	}
+
+	return sum
+}
+
+func TestCosineDistanceCorrelation(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	client, _, cleanup := InitServerConnection(ctx, t)
+	defer cleanup()
+
+	for _, model := range libraryEmbedModels {
+		testCases := []struct {
+			a string
+			b string
+			c string
+		}{
+			{"cat", "kitten", "dog"},
+			{"king", "queen", "baron"},
+			{"paris", "london", "vancouver"},
+			{"The cat is sleeping on the sofa", "A feline is sleeping on the couch", "Quantum physics is complex"},
+			{"I love programming in python", "Coding in python brings me joy", "Pizza is delicious"},
+			{"Machine learning is fascinating", "Artificial intelligence is amazing", "I need to buy groceries"},
+			{"The quick brown fox jumps over the lazy dog", "A fast brown fox leaps over a sleepy dog", "The weather is warm and sunny today"},
+		}
+
+		for _, tc := range testCases {
+			testEmbed := make(map[string][]float32)
```

Comment:
It might be useful to inject a `t.Run(model, func(t *testing.T) {` in here so the discrete models could be tested with a `-run TestCosineDistanceCorrelation/embeddinggemma`